src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ARRAY_SIZE = 10;
+    private int arrayListSize;
+    private T[] listsOfElements = (T[]) new Object[MAX_ARRAY_SIZE];
+
     @Override
     public void add(T value) {
-
+        if (arrayListSize == listsOfElements.length) {
+            grow();
+        }
+        listsOfElements[arrayListSize++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > arrayListSize) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound exception"
+                    + index);
+        }
+        grow();
+        System.arraycopy(listsOfElements, index, listsOfElements,
+                index + 1, arrayListSize - index);
+        listsOfElements[index] = value;
+        arrayListSize++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) listsOfElements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        listsOfElements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = listsOfElements[index];
+        final int newSize;
+        if ((newSize = arrayListSize - 1) >= index) {
```

Comment:
do we need this variable(`newSize`) and this check(`if ((newSize = arrayListSize - 1) >= index) {`)?